### DISCUSSION TITLE
init.cc:1256 NCCL WARN Your program may be hanging, after validation. how to fix it

### DISCUSSION THREAD

**Discussion by Devoe-97** [Lightning-AI/pytorch-lightning#8213] [DDP / multi-GPU / multi-node]
when i run my code in one gpu, it seems ok.
when i run my code in multi gpus, it's training is ok but the nccl error occur after validation and saving model
![image](https://user-images.githubusercontent.com/32172150/123896718-a6195f80-d994-11eb-8291-aeb099174991.png)
how should i fix it?

**Comment by thecooltechguy** [comment]
Hi, I often faced similar NCCL hanging issues, if there wasn't enough data for all dataloader workers for all gpus for both training and validation data.

**Comment by Borda** [comment]
Could you pls provide a sample code to reproduce, also some more env details? :rabbit: 
cc: @awaelchli

**Comment by awaelchli** [comment]
Have you followed the instructions on the screen there? It says adding `NCCL_DEBUG=INFO` to get the full logs from NCCL.
